tests/testthat/test-preProcessWorks.R

```diff
@@ -90,21 +91,156 @@ test_that("preProcess works when provides url, archive, targetfile and destinati
   on.exit({
     testOnExit(testInitOut)
   }, add = TRUE)
-  ras <- reproducible::preProcess("https://github.com/tati-micheletti/host/raw/master/data/rasterTest.zip",
+  ras <- reproducible::preProcess(url = "https://github.com/tati-micheletti/host/raw/master/data/rasterTest.zip",
                                   targetFile = "rasterTest.tif",
                                   archive = "rasterTest.zip",
                                   destinationPath = tmpdir)
   testthat::expect_is(object = ras, class = "list")
   testthat::expect_true(file.exists(ras$targetFilePath))
 })
+
 test_that("preProcess works when provides url, targetfile and destinationPath", {
   testInitOut <- testInit("raster", needGoogle = FALSE)
   on.exit({
     testOnExit(testInitOut)
   }, add = TRUE)
-  ras <- reproducible::preProcess("https://github.com/tati-micheletti/host/raw/master/data/rasterTest.zip",
+  ras <- reproducible::preProcess(url = "https://github.com/tati-micheletti/host/raw/master/data/rasterTest.zip",
                                   targetFile = "rasterTest.tif",
                                   destinationPath = tmpdir)
   testthat::expect_is(object = ras, class = "list")
   testthat::expect_true(file.exists(ras$targetFilePath))
 })
+test_that("preProcess works when provides url and destinationPath for a .rar file", {
+  testInitOut <- testInit("raster", needGoogle = FALSE)
+  on.exit({
+    testOnExit(testInitOut)
+  }, add = TRUE)
+  OSystem <- Sys.info()["sysname"]
```

Comment:
looks like a lot of duplicated code from the main unrar section -- see prev comments re: `Sys.which()` and wrap this duplicated bit in an internal function (use `@keywords internal` and don't `@export` in the docs)